tests/integration/WC_Facebookcommerce_Integration_Test.php

```diff
@@ -771,8 +773,396 @@ public function test_is_product_sync_enabled_filter() {
 		$this->assertFalse( $this->integration->is_product_sync_enabled() );
 	}
 
+
+	/** @see \WC_Facebookcommerce_Integration::on_product_delete() */
+	public function test_on_product_delete_with_simple_product() {
+
+		$product = $this->tester->get_product( [ 'status' => 'publish' ] );
+
+		$integration = $this->make( \WC_Facebookcommerce_Integration::class, [
+			'delete_product_item'  => Expected::once(),
+			// called from delete_product_item() and from delete_product_group()
+			'get_product_fbid'     => null,
+		] );
+
+		$integration->on_product_delete( $product->get_id() );
+	}
+
+
+	/** @see \WC_Facebookcommerce_Integration::on_product_delete() */
+	public function test_on_product_delete_with_variable_product() {
+
+		$product = $this->tester->get_variable_product( [
+			'children' => 3,
+			'status'   => 'publish',
+		] );
+
+		$integration = $this->make( \WC_Facebookcommerce_Integration::class, [
+			'delete_product_item' => Expected::never(),
+			// called from delete_product_group()
```

Comment:
Should we maybe explicitly expect `delete_product_group` to be called?

Reply:
Codeception wouldn't let me because `delete_product_group()` is private. It works with `delete_product_item()` because that one doesn't have the accessibility defined.

Do you know if there is a way to bypass that restriction? If not, I think I can add a note to explain.

Reply:
I added a note about why we are expecting that `get_product_fbid()` is called instead of using `delete_product_group()`: 92a9356.